src/BrightScriptDebugSession.ts

```diff
@@ -550,11 +582,18 @@ export class BrightScriptDebugSession extends DebugSession {
                 //we found multiple files with the exact same path (unlikely)...nothing we can do about it.
             }
         }
-        //use debugRootDir if provided, or rootDir if not provided.
-        let rootDir = this.launchArgs.debugRootDir ? this.launchArgs.debugRootDir : this.launchArgs.rootDir;
+        let rootDir = this.launchArgs.sourceDirs ? this.launchArgs.sourceDirs : this.launchArgs.rootDir;
```

Comment:
`sourceDirs` is an array, but `rootDir` is a single string. So a `for of` loop on a string will loop through every character in the string. Please test to verify, but I think you need to wrap `this.launchArgs.rootDir` in brackets to put it into an array.

`let rootDir = this.launchArgs.sourceDirs ? this.launchArgs.sourceDirs : [this.launchArgs.rootDir];`